src/logger.ts

```diff
@@ -2,6 +2,27 @@ import { createLogger, format, transports } from "winston";
 import { getRequestId } from "./customExpressHttpContext";
 import fs from "fs";
 const logDir = "logs";
+
+// For now, we only support three log levels
+export type LogLevel = "error" | "info" | "debug";
+export const supportedLogLevels: LogLevel[] = ["error", "info", "debug"];
```

Comment:
You can restructure these repeated strings by having a LogLevelInfo class which could look something like:
```
enum LogLevel {
Info = "info",
Error = "error",
Debug = "debug"
}

class LogLevelInfo {

public static Error : LogLevelInfo = new LogLevelInfo(LogLevel.Error, 0);
public static Info : LogLevelInfo = new LogLevelInfo(LogLevel.Info, 2);
public static Debug : LogLevelInfo = new LogLevelInfo(LogLevel.Debug, 4);

private constructor(public readonly logLevel: LogLevel, public readonly order: number) {}

static tryParse(logLevel: string):LogLevelInfo { 
// returns null if level does not exist
}
}

Reply:
Refactored with Enums; clean enough, but it doesn't seem to be possible to stay fully type-safe, unfortunately :/